EventFilter/RPCRawToDigi/interface/RPCTwinMuxRecord.icc

```diff
@@ -0,0 +1,775 @@
+{
+    record_ &= ~partition_mask_;
+    record_ |= (((::uint32_t)(_partition) << partition_offset_) & partition_mask_);
+}
+
+inline void RPCLinkRecord::setPartitionData(::uint8_t _data)
+{
+    record_ &= ~partition_data_mask_;
+    record_ |= (((::uint32_t)(_data) << partition_data_offset_) & partition_data_mask_);
+}
+
+inline void RPCBXRecord::set(::uint8_t const _record)
+{
+    record_ = _record;
+}
+
+inline void RPCBXRecord::reset()
+{
+    record_ = 0x0;
+}
+
+inline ::uint8_t const & RPCBXRecord::getRecord() const
+{
+    return record_;
+}
+
+inline bool RPCBXRecord::isBC0() const
+{
+    return ((record_ & bc0_mask_) ? true : false);
+}
+
+inline unsigned int RPCBXRecord::getBXCounter() const
+{
+    return (record_ & bcn_mask_) >> bcn_offset_;
+}
+
+inline void RPCBXRecord::setBC0(bool _bc0)
+{
+    if (_bc0)
+        record_ |= bc0_mask_;
+    else
+        record_ &= ~(bc0_mask_);
+}
+
+inline void RPCBXRecord::setBXCounter(unsigned int _bcn)
+{
+    record_ &= ~bcn_mask_;
+    record_ |= (((::uint8_t)(_bcn) << bcn_offset_) & bcn_mask_);
+}
+
+inline void RPCRecord::set(unsigned int _nword, ::uint64_t const _word)
+{
+    if (_nword < 2)
+        record_[_nword] = _word;
+}
+
+inline void RPCRecord::reset()
+{
+    record_[0] = TwinMuxRecord::rpc_first_identifier_
+        | ((::uint64_t)RPCLinkRecord::da_mask_ << link_record_0_offset_)
+        | ((::uint64_t)RPCLinkRecord::da_mask_ << link_record_1_offset_);
+    record_[1] = TwinMuxRecord::rpc_second_identifier_
+        | ((::uint64_t)RPCLinkRecord::da_mask_ << link_record_2_offset_)
+        | ((::uint64_t)RPCLinkRecord::da_mask_ << link_record_3_offset_)
+        | ((::uint64_t)RPCLinkRecord::da_mask_ << link_record_4_offset_);
+}
+
+inline ::uint64_t const * RPCRecord::getRecord() const
+{
+    return record_;
+}
+
+inline int RPCRecord::getBXOffset() const
+{
+    return -4 + (int)((record_[0] & bx_offset_mask_) >> bx_offset_offset_);
+}
+
+inline bool RPCRecord::hasOverflow() const
+{
+    return ((record_[0] & overflow_mask_) ? true : false);
+}
+
+inline RPCBXRecord RPCRecord::getRPCBXRecord(unsigned int _link) const
+{
+    RPCBXRecord _bx_record;
+    switch (_link) {
+    case 0:
+        _bx_record.set((record_[0] >> bx_record_0_offset_) & bx_record_mask_);
+        break;
+    case 1:
+        _bx_record.set((record_[0] >> bx_record_1_offset_) & bx_record_mask_);
+        break;
+    case 2:
+        _bx_record.set((record_[0] >> bx_record_2_offset_) & bx_record_mask_);
+        break;
+    case 3:
+        _bx_record.set((record_[0] >> bx_record_3_offset_) & bx_record_mask_);
+        break;
+    case 4:
+        _bx_record.set((record_[0] >> bx_record_4_offset_) & bx_record_mask_);
+        break;
+    }
+    return _bx_record;
+}
+
+inline RPCLinkRecord RPCRecord::getRPCLinkRecord(unsigned int _link) const
+{
+    RPCLinkRecord _link_record;
+    switch (_link) {
+    case 0:
+        _link_record.set((record_[0] >> link_record_0_offset_) & link_record_mask_);
```

Comment:
switch-based implementation here seems excessive.
```link_record.set((record_[link_record_index[link]] >> link_record_offse[link]) & link_record_mask);```

is quite a bit shorter

Similar comment applies to some other switch statements in this PR